pkg/controllers/override/imagepatch.go

```diff
@@ -0,0 +1,203 @@
+
+	// parse patches and store the new image values in patchMap
+	for i := range imageOverriders {
+		imageOverrider := imageOverriders[i]
+		if imageOverrider.ImagePath != "" {
+			if err := parsePatchesFromImagePath(fedObject, &imageOverrider, patchMap); err != nil {
+				return nil, err
+			}
+		} else {
+			if err := parsePatchesFromWorkload(fedObject, &imageOverrider, patchMap); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// convert patchMap to patchList and return
+	patches := make(fedcorev1a1.OverridePatches, len(patchMap))
+	index := 0
+	for path, value := range patchMap {
+		patches[index] = fedcorev1a1.OverridePatch{
+			Op:   OperatorReplace,
+			Path: path,
+			// OverridePatch's value is apiextensionsv1.JSON type, which raw should be wrapped with "".
+			Value: apiextensionsv1.JSON{Raw: []byte(fmt.Sprintf("\"%s\"", value))},
+		}
+		index++
+	}
+
+	return patches, nil
+}
+
+// parsePatchesFromImagePath applies overrider to image value on the specified path,
+// and generates a patch which is stored in patchMap
+func parsePatchesFromImagePath(
+	fedObject fedcorev1a1.GenericFederatedObject,
+	imageOverrider *fedcorev1a1.ImageOverrider,
+	patchMap map[string]string,
+) error {
+	imagePath := imageOverrider.ImagePath
+	imageValue := patchMap[imagePath]
+
+	// get image value
+	if imageValue == "" {
+		// get source obj
+		sourceObj, err := fedObject.GetSpec().GetTemplateAsUnstructured()
+		if err != nil {
+			return fmt.Errorf("failed to get sourceObj from fedObj: %w", err)
+		}
+
+		imageValue, err = ParseValueFromUnstructuredObj(sourceObj, imageOverrider.ImagePath)
+		if err != nil {
+			return fmt.Errorf("failed to parse image value from unstructured obj: %w", err)
+		}
+	}
+
+	// apply image override
+	newImageValue, err := applyImageOverride(imageOverrider.ImagePath, imageValue, imageOverrider)
+	if err != nil {
+		return fmt.Errorf("failed apply image override: %w", err)
+	}
+	patchMap[imagePath] = newImageValue
+
+	return nil
+}
+
+// parsePatchesFromWorkload applies overrider to image value on the default path of workload,
+// and generates a patch which is stored in patchMap
+func parsePatchesFromWorkload(
+	fedObject fedcorev1a1.GenericFederatedObject,
+	imageOverrider *fedcorev1a1.ImageOverrider,
+	patchMap map[string]string,
+) error {
+	// get pod spec from fedObj
+	sourceObj, err := fedObject.GetSpec().GetTemplateAsUnstructured()
+	if err != nil {
+		return fmt.Errorf("failed to get sourceObj from fedObj: %w", err)
+	}
+	gvk := sourceObj.GetObjectKind().GroupVersionKind()
+	podSpec, err := podutil.GetResourcePodSpec(fedObject, gvk)
+	if err != nil {
+		return fmt.Errorf("failed to get podSpec from sourceObj: %w", err)
+	}
+
+	// convert user-specified containerNames into set
+	specifiedContainers := sets.New[string]()
+	for _, name := range imageOverrider.ContainerNames {
+		if name != "" {
+			specifiedContainers.Insert(name)
+		}
+	}
+
+	for index, container := range podSpec.Containers {
```

Comment:
Did we forget to include init containers?

Reply:
The relevant code has been added.